aspnetcore/signalr/scale.md

```diff
@@ -104,14 +104,85 @@ The preceding conditions make it likely to hit the 10 connection limit on a clie
 
 ## Linux with Nginx
 
-Set the proxy's `Connection` and `Upgrade` headers to the following for SignalR WebSockets:
+The following contains the minimum required settings to enable WebSockets, ServerSentEvents, and LongPolling for SignalR:
 
 ```nginx
-proxy_set_header Upgrade $http_upgrade;
-proxy_set_header Connection $connection_upgrade;
+http {
+  map $http_upgrade $connection_upgrade {
+    default Upgrade;
+    '' keep-alive;
```

Comment:
What's the reason for basing the Connection header on the incoming Upgrade header? Maybe the following which bases it on the incoming Connection header would make more sense. It seems more straightforward not to look at a different header.

```
map $http_connection $connection_upgrade {
  Upgrade Upgrade;
  default keep-alive;
}
```

Reply:
Doesn't work if the request has a multi-value Connection header.

Reply:
Looks like it might work if I do `"~*Upgrade" Upgrade;`

Reply:
Given that, I'm fine with the way you have it now or the old way.

Reply:
What does `"~*Upgrade" Upgrade;` mean?

Reply:
Case-insensitive check for "Upgrade" being in the request header. For example, Firefox sets "keep-alive, Upgrade" which wouldn't match normally.